bench/irmin-pack/tree.ml

```diff
@@ -470,20 +909,17 @@ struct
           Store.freeze repo ~max:[ c ] ~min_upper:[ c ]
         else Lwt.return_unit
       in
-      let* () =
-        if i = ncommits - 1 then Store.PrivateLayer.wait_for_freeze repo
-        else Lwt.return_unit
-      in
       (* Something else than pause could be used here, like an Lwt_unix.sleep
          or nothing. See #1293 *)
       Lwt.pause ()
     in
+    let on_end _i = Store.PrivateLayer.wait_for_freeze repo in
```

Comment:
you can use a unit instead of int for `on_end` I think

Reply:
👍